### PR TITLE
Migrate to DB_URL, add caching in dockerbuilds

### PR DESCRIPTION
**Pull request by Shubhaankar-Sharma**
## Checklist

- [ ] If endpoints were changed then they have been documented and tested.
    - [ ] I have updated the docmentation to reflect the changes.
    - [ ] I have updated the tests to support the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new endpoint or parameter).
- [x] This PR is a breaking change (e.g. endpoint or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by Shubhaankar-Sharma** [comment]
:mhm: m7 when???

File: .github/workflows/lint-test.yml [hunk 1]
```diff
@@ -66,22 +66,14 @@ jobs:
       - name: Migrate
         run: go run main.go migrate_up
         env:
-          POSTGRES_USER: root
-          POSTGRES_PASSWORD: mysecret
-          DB_NAME: cdn
-          DB_HOST: localhost
-          DB_PORT: 5432
+          DB_URI: postgres://root:mysecret@localhost:5432/cdn?sslmode=disable
```
Comment (SylteA) [inline .github/workflows/lint-test.yml]: Can we please use constants for this over all repositories?

postgres://twt:twt@postgres:5432/twt

File: codecov.yml [hunk 2]
```diff
@@ -4,11 +4,11 @@ codecov:
 coverage:
   precision: 2
   round: up
-  range: "60...100"
+  range: "50...100"
   status:
     patch:
       default:
-        target: 60%
+        target: 50%
```
Comment (SylteA) [inline codecov.yml:11]: target slowly decreasing :3

**Review by SylteA — CHANGES_REQUESTED** [review]

**Review by SylteA — APPROVED** [review]

**Comment by Shubhaankar-Sharma** [comment]
m7md let me know when you update credentials

**Pull request merged**